docs/docs/integrations/tools/goat.ipynb

```diff
@@ -0,0 +1,250 @@
+    "\n",
+    "# Initialize regular Solana wallet\n",
+    "keypair = Keypair.from_base58_string(os.getenv(\"SOLANA_WALLET_SEED\") or \"\")\n",
+    "wallet = solana(client, keypair)\n",
+    "\n",
+    "tools = get_on_chain_tools(\n",
+    "        wallet=wallet,\n",
+    "        plugins=[\n",
+    "            send_solana(),\n",
+    "            spl_token(SplTokenPluginOptions(\n",
+    "                network=\"mainnet\",  # Using devnet as specified in .env\n",
+    "                tokens=SPL_TOKENS\n",
+    "            )),\n",
+    "        ],\n",
+    "    )\n",
+    "```\n",
+    "\n",
+    "## Invocation\n",
+    "```python\n",
+    "tools[\"get_balance\"].invoke({ \"address\": \"0x1234567890123456789012345678901234567890\" })\n",
+    "```\n",
+    "\n",
+    "## Use within an agent\n",
+    "\n",
+    "```python\n",
+    "import os\n",
+    "import asyncio\n",
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "# Load environment variables\n",
+    "load_dotenv()\n",
+    "\n",
+    "from langchain_openai import ChatOpenAI\n",
+    "from langchain.agents import AgentExecutor, create_tool_calling_agent\n",
+    "from langchain_core.prompts import ChatPromptTemplate\n",
+    "from solana.rpc.api import Client as SolanaClient\n",
+    "from solders.keypair import Keypair\n",
+    "\n",
+    "from goat_adapters.langchain import get_on_chain_tools\n",
+    "from goat_wallets.solana import solana, send_solana\n",
+    "from goat_plugins.spl_token import spl_token, SplTokenPluginOptions\n",
+    "from goat_plugins.spl_token.tokens import SPL_TOKENS\n",
+    "\n",
+    "# Initialize Solana client\n",
+    "client = SolanaClient(os.getenv(\"SOLANA_RPC_ENDPOINT\"))\n",
+    "\n",
+    "# Initialize regular Solana wallet\n",
+    "keypair = Keypair.from_base58_string(os.getenv(\"SOLANA_WALLET_SEED\") or \"\")\n",
+    "wallet = solana(client, keypair)\n",
+    "\n",
+    "# Initialize LLM\n",
+    "llm = ChatOpenAI(model=\"gpt-4o-mini\")\n",
+    "\n",
+    "def main():\n",
+    "    \n",
+    "    # Get the prompt template\n",
+    "    prompt = ChatPromptTemplate.from_messages(\n",
+    "        [\n",
+    "            (\"system\", \"You are a helpful assistant\"),\n",
+    "            (\"placeholder\", \"{chat_history}\"),\n",
+    "            (\"human\", \"{input}\"),\n",
+    "            (\"placeholder\", \"{agent_scratchpad}\"),\n",
+    "        ]\n",
+    "    )\n",
+    "\n",
+    "    # Initialize tools with Solana wallet\n",
+    "    tools = get_on_chain_tools(\n",
+    "        wallet=wallet,\n",
+    "        plugins=[\n",
+    "            send_solana(),\n",
+    "            spl_token(SplTokenPluginOptions(\n",
+    "                network=\"mainnet\",  # Using devnet as specified in .env\n",
+    "                tokens=SPL_TOKENS\n",
+    "            )),\n",
+    "        ],\n",
+    "    )\n",
+    "\n",
+    "    agent = create_tool_calling_agent(llm, tools, prompt)\n",
```

Comment:
Could you remove the agent part? The tool should just show the tool. I know not all integration pages do that, but we don't want duplication in the docs

Reply:
FWIW the agent implementation is using very old langchain code. Agents should be written using langgraph prebuilts these days

Reply:
Done!